clearly-defined/src/main/kotlin/ClearlyDefinedService.kt

```diff
@@ -409,6 +410,19 @@ interface ClearlyDefinedService {
     @POST("definitions")
     fun getDefinitions(@Body coordinates: Collection<String>): Call<Map<String, Defined>>
 
+    /**
+     * Search for existing definitions based on the [pattern] string provided. The pattern string should contain
+     * the parts of the coordinates (typically namespace, name, and version) relevant for the search. Result is a
+     * list with the ClearlyDefined URIs to all the definitions that are matched by the pattern.
+     *
+     * Note that the _/definitions_ GET endpoint behaves differently depending on the parameters passed in. If only
+     * a pattern string is provided, result is a list of URIs - this use case is covered by this function. It is
+     * also possible to provide other parameters like type, provider, namespace, name, etc. Then the endpoint
+     * returns partly filled definition documents.
```

Comment:
IMO it's confusing that you seem to describe parameters that the `definitions` entry point theoretically supports, but this function does not. You should only describe what *this* function supports / does.

Also, I'm missing a link to the upstream API docs like we have for the other functions.

Reply:
Reworded and added the link. The comment now only describes the relevant part for this function.

I have to admit, I was also confused by the API of ClearlyDefined in this case. Actually, there are two different functionalities behind the definitions GET endpoint:
* You can pass in a pattern string and get back a list of matching URIs or
* you can pass in a number of parameters defining package coordinates and get back partly filled definition objects.

As we only use the first functionality, I stripped down the function comment to this use case as requested.